src/Microsoft.AspNet.Mvc.Core/Formatters/MediaTypeComponent.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Internal;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNet.Mvc.Formatters
+{
+    /// <summary>
+    /// Represents a component within a media type.
+    /// </summary>
+    public struct MediaTypeComponent : IEquatable<MediaTypeComponent>
+    {
+        /// <summary>
+        /// The name of the type in a media type when represented as a <see cref="MediaTypeComponent"/>.
+        /// </summary>
+        public static readonly StringSegment Type = new StringSegment("type");
+
+        /// <summary>
+        /// The name of the subtype in a media type when represented as a <see cref="MediaTypeComponent"/>.
+        /// </summary>
+        public static readonly StringSegment Subtype = new StringSegment("subtype");
```

Comment:
I think it's really strange to represent the 'type' and 'subtype' as part of an Enumerator. Every media type has them, why not simplify this and simplify your life?